### PR TITLE
Hide nonce implementation details in metadata

### PR DESCRIPTION
**Pull request by re-gius**
See https://github.com/polkadot-fellows/runtimes/issues/248 : using `TypeWithDefault` having derived `TypeInfo` for `Nonce` causes a breaking change in metadata for nonce type because it's no longer `u64`.

Adding a default implementation of `TypeInfo` for `TypeWithDefault` to restore the original type info in metadata.

### CODE REVIEW

**Comment by re-gius** [comment]
What metadata shows on `solochain` local node

```
{
    name: Nonce
    type: 4
}
[...]
{
  id: 4
  type: {
    path: []
    params: []
    def: { Primitive: U32 }
    docs: []
  }
}
```

File: substrate/primitives/runtime/src/type_with_default.rs [hunk 1]
```diff
@@ -50,6 +50,17 @@ impl<T, D: Get<T>> TypeWithDefault<T, D> {
 	}
 }
 
+/// Hides implementation details from the outside (for metadata type information).
+///
+/// The type info showed in metadata is the one of the inner value's type.
```
Comment (kianenigma) [inline substrate/primitives/runtime/src/type_with_default.rs:55]:
```suggestion
// Hides implementation details from the outside (for metadata type information).
//
// The type info showed in metadata is the one of the inner value's type.
```

File: substrate/primitives/runtime/src/type_with_default.rs [hunk 2]
```diff
@@ -40,7 +40,7 @@ use serde::{Deserialize, Serialize};
 /// A type that wraps another type and provides a default value.
 ///
 /// Passes through arithmetical and many other operations to the inner value.
-#[derive(Encode, Decode, TypeInfo, Debug, MaxEncodedLen)]
+#[derive(Encode, Decode, Debug, MaxEncodedLen)]
```
Comment (kianenigma) [inline substrate/primitives/runtime/src/type_with_default.rs:43]: Should update the doc of the type to specify this nuance.

**Comment by kianenigma** [comment]
The only minor downside of this is that now we would have to wait for the next release, and then update the fellowship repo as well

Set your reminders @re-gius  :)

**Review by kianenigma — APPROVED** [review]

File: substrate/primitives/runtime/src/type_with_default.rs [hunk 3]
```diff
@@ -565,5 +577,11 @@ mod tests {
 		}
 		type U128WithDefault = TypeWithDefault<u128, Getu128>;
 		impl WrapAtLeast32Bit for U128WithDefault {}
+
+		assert_eq!(U8WithDefault::type_info(), <u8 as TypeInfo>::type_info());
```
[the remark this answers is not in the document]
Comment (re-gius) [reply substrate/primitives/runtime/src/type_with_default.rs:582]: Exactly, I just double-checked: it does not pass with the derived `TypeInfo`

**Review by jsdw — APPROVED** [review]
LGTM!

**Pull request merged**